### PR TITLE
Update example_config.py.

### LINKED ISSUE #331: Update ExampleConfig to be more like the ocean.js ConfigHelper
ocean.js offers the `ConfigHelper.getConfig` method which takes a network name, and returns a dictionary of config parameters for the corresponding network like chainId, provider URL, aquarius URL, etc.  

ocean.py offers the `ExampleConfig.get_config` method that returns a `Config` instance that is used by unit tests but wouldn't work for a real network. 

This issue is to update the ocean.py `ExampleConfig` to be more like the ocean.js `ConfigHelper`.

See for details: 
1. https://github.com/oceanprotocol/ocean.js/blob/main/src/utils/ConfigHelper.ts
2. https://github.com/oceanprotocol/ocean.py/blob/main/ocean_lib/example_config.py

### PR DESCRIPTION
**Pull request by mariacarmina**
Fixes #331  .

Changes proposed in this PR:

- added config support for ganache, ropsten, rinkeby, mainnet, polygon-mainnet and BSC;
- removed unnecessary uni test that worked only for ganache network from `test_config.py`;
- refactored `config_defaults` from `config.py` to be able to use it as a basic config structure.

### CODE REVIEW

**Comment by mariacarmina** [comment]
As I saw in #393 should I also add the store intervals for each network as well, @DMats @calina-c @alexcos20 ?

**Comment by mariacarmina** [comment]
I have created a draft refactor related to `example_config.py` that matches some discussed aspects. I missed to integrate the network name in the config dict, but as far as I saw the `network` key from the ocean.py config stands for the actual RPC, not for the network name itself. I fetched the chain id of that env var RPC and then I searched through the chain ids from `CONFIG_HELPER_NETWORKS` dict. I do not find useful in this situation the `network name`, since we search for the chain id. Even if in ocean.js the search is made through the network name, in ocean.py we provide different concepts for config object.

**Review by DMats — CHANGES_REQUESTED** [review]

File: ocean_lib/example_config.py [hunk 1]
```diff
@@ -4,47 +4,140 @@
...
-            "resources": {
-                "metadata_cache_uri": "http://aquarius:5000",
-                "provider.url": "http://localhost:8030",
-                "provider.address": "0x068Ed00cF0441e4829D9784fCBe7b9e26D4BD8d0",
-                "downloads.path": "consume-downloads",
-            },
-        }
+CONFIG_HELPER_NETWORKS = {
+    1337: config_defaults,
+    3: {
```
[the remark this answers is not in the document]
Comment (mariacarmina) [reply ocean_lib/example_config.py:21]: I have tried before with `dict.update()`, but it modifies the object itself, not returning a new one and in the end all the networks will have the same config.

**Comment by mariacarmina** [comment]
I have added the NETWORK_URL env var to `pytest.ini` and also to `tox.ini` (`127.0.0.1:8545` and also tested with `172.15.0.3:8545` which is present in the `barge` script) and some tests failed due to the fact that Aquarius could not fetch the created ddo from the unit tests like so 
Screenshot from 2021-08-06 15-45-03
And I have seen other error messages such as `The did <did> is not in the OceanDB.`

**Pull request closed without merging**